esphome/components/sdm_meter/sensor.py

```diff
@@ -74,19 +81,27 @@
             cv.Optional(CONF_PHASE_B): PHASE_SCHEMA,
             cv.Optional(CONF_PHASE_C): PHASE_SCHEMA,
             cv.Optional(CONF_FREQUENCY): sensor.sensor_schema(
-                UNIT_HERTZ, ICON_CURRENT_AC, 3
+                UNIT_HERTZ, ICON_CURRENT_AC, 3, DEVICE_CLASS_EMPTY, STATE_CLASS_NONE
```

Comment:
This is a measurement in time and you might want the min/max of this
```suggestion
                UNIT_HERTZ, ICON_CURRENT_AC, 3, DEVICE_CLASS_EMPTY, STATE_CLASS_MEASUREMENT
```